modules/gapi/include/opencv2/gapi/infer/ov.hpp

```diff
@@ -691,6 +691,29 @@ namespace wip { namespace ov {
  */
 struct benchmark_mode { };
 
+struct workload_type {
+    using callback = std::function<void(const std::string &type)>;
+    using listener = std::pair<int, callback>;
+    std::shared_ptr<void> addListener(callback cb){
+        int id = nextId++;
+        listeners.emplace_back(id, std::move(cb));
+
+        auto remover = [this, id](void*){ removeListener(id);};
+
+        return std::shared_ptr<void>(nullptr, remover);
```

Comment:
I don't think it is a good idea to remove listeners that way. It seems a bit tricky. I understand that this variable may outlive G-API OV backend. But, might be, we can clean this variable's listeners in the destructor of G-API OV backend itself. Please note that we are the "users" of given API to add listeners and we 100% know when we add a listener and when we can remove it, so there is no need for some RAII idiom for remove. If this is allowable, we can preserve these compile args somewhere in backend to access variable in the destructor.